### PR TITLE
AWS - S3 Bucket Policies - closes #431

### LINKED ISSUE #431: AWS - S3 Bucket Policies

### PR DESCRIPTION
**Pull request by Matt25969**

### CODE REVIEW

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# S3
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]: # S3 Bucket Policies

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	- [Bucket Privacy](#bucket-privacy)
+	- [Security](#security)
+- [Tasks](#tasks)
+
+<!--TOC_END-->
+## Overview
+
+S3 is a Foundation Tool in the AWS Eco-System, it is a Highly Available Object Store that allows 'infinite' storage, you will never run out of space to store your files.
+
+## Policies
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]:
```suggestion
```

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+- [Overview](#overview)
+- [Policies](#policies)
+	- [Bucket Privacy](#bucket-privacy)
+	- [Security](#security)
+- [Tasks](#tasks)
+
+<!--TOC_END-->
+## Overview
+
+S3 is a Foundation Tool in the AWS Eco-System, it is a Highly Available Object Store that allows 'infinite' storage, you will never run out of space to store your files.
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]:
```suggestion
```

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## Policies
+
+### Bucket Privacy
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]:
```suggestion
## Bucket Privacy
```

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+By default, S3 buckets are private, this means that only the root user has access to the bucket.  For the majority of use cases this is sufficient, as the buckets will normally be used to save data from other applications.
+
+Giving an S3 public access means that anybody with the URL for the bucket can access its contents, for hosting a static website this is ideal, however when you need to keep data private you need to keep your bucket private!
+
+You can also control access to your Bucket, using these means you can set who has access, and what access they have.
+
+```
+aws_s3_intro_1.png
+```
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]: What's this for?

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+### Security
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]:
```suggestion
## Security
```

So which should I use? ACL or Policies?
ACLs seem to have more basic access control?

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+A bucket policy is where we can define explicit access or denial of access to our data at a bucket level.  These polices are written using JSON, meaning that we have very configurable policies with granular access.
+
+An ACL is how we can give explicit access or denial of access to our data for users from outside of our own AWS Accounts.  The permissions we can add are quite broad for example 'Write Objects'.
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]:
```suggestion
An Access Control List (ACL) is how we can give explicit access or denial of access to our data for users from outside of our own AWS Accounts.
The permissions we can add are quite broad for example `Write Objects`.
```

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+Navigate back to the **Overview** tab of the bucket.
+
+Click the **Permissions** tab.
+
+Now click the **Bucket Policy** button.
+
+Copy the code below into the Bucket policy editor, ensure that you change the name of the bucket to reflect the current bucket you are working with.
+
+
+```
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]:
```suggestion
```JSON
```

File: topics/aws/modules/s3-bucket-policies/README.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+         "Resource": "arn:aws:s3:::<name_of_your_bucket>/*"
+      }
+  ]
+}
+```
+Click the blue **Save** button.
+
+Again navigate to the **Object URL** Link you tried earlier, you should see now that the file is rendered by your browser.
+
+The last step is to undo everything we have done, you can delete the bucket policy, Block all Public Access or simply delete the bucket.  Ensure that the contents of your bucket are once again private before moving on.
```
Comment (bob-crutchley) [inline topics/aws/modules/s3-bucket-policies/README.md]: Could you split this into headings?

File: topics/aws/modules/s3-bucket-policies/README.md (hunk 5 above)
Comment (Matt25969) [reply topics/aws/modules/s3-bucket-policies/README.md]: I dont know how to do images yet - im am concerned about the time it takes to add images versus the speed we are generating material.

**Comment by tvaidotas** [comment]
closing this for the time being

**Pull request closed without merging**